Library/Homebrew/test/api/internal_tap_json/formula_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+describe "Internal Tap JSON -- Formula" do
+  let(:internal_tap_json) { File.read(TEST_FIXTURE_DIR/"internal_tap_json/homebrew-core.json").chomp }
+  let(:tap_git_head) { "9977471165641744a829d3e494fa563407503297" }
+
+  context "when generating JSON", :needs_macos do
+    before do
+      cp_r(TEST_FIXTURE_DIR/"internal_tap_json/homebrew-core", Tap::TAP_DIRECTORY/"homebrew")
```

Comment:
Sometimes it seems like some of these files don't get copied over or get deleted before the tests can run. I'm not sure why that is.

Reply:
@apainintheneck is that only when running in parallel or can you reproduce it serially?

Reply:
It doesn't fail when running one file but does when running the whole test suite. I'll try running everything serially to see what happens.

Reply:
Everything passes when running the tests serially locally.

Reply:
Yeh, smells like this is just multiple tests creating/deleting the same files at the same time. May need to have a randomised/test runner based filename/directory name.

Reply:
Interesting... my mental model of how the tests work is that each process would have a separate temporary directory and wouldn't pollute each others workspace. The test directories also seem to get cleaned up after each test.

Reply:
Only if you're creating a new e.g. tmpdir for each. Otherwise: use same filename in multiple tests, run them in parallel and they will happily cleanup the WIP of another test.

Reply:
That seems like a major bug in the test suite then, doesn't it? We should be able to create one temp directory per process and that would allow for better reproducibility in that case.

Reply:
It looks like each process gets a temporary directory already which is very good. The code that makes that work is here.

https://github.com/Homebrew/brew/blob/c35546167da22c3c00d34332ff84a1bb40153766/Library/Homebrew/test/support/lib/startup/config.rb#L8-L16

Reply:
Yeh, there's definitely some temporary handling but there's also definitely ways even with that that you can inadvertently add/remove files in different tests running in parallel (this has bitten me before). Good luck!